integration-tests/cstor-pool/cstor_integration_test.go

```diff
@@ -0,0 +1,370 @@
+			cspCount := isHealthyCspCount(spcObj.Name, 3)
+			Expect(cspCount).To(Equal(3))
+		})
+	})
+
+	//Test Case #3 : Dependent on above test case #1 . | TestType : Reconciliation
+	When("We delete 3 pool out of 3 by deleting 3 csps", func() {
+		It("3 new pool should come up again", func() {
+			clientInstance.deleteCSP(spcObj.Name, 3)
+			// We expect 3 cstorPool objects.
+			cspCount := isHealthyCspCount(spcObj.Name, 3)
+			Expect(cspCount).To(Equal(3))
+		})
+	})
+
+})
+
+
+var _ = Describe("RAIDZ2 SPARSE SPC", func() {
+
+	var (
+		spcObj *apis.StoragePoolClaim
+	)
+	BeforeEach(func() {
+		spcObj = spc.NewSPCBuilder().
+			WithName("sparse-raidz2-auto").
+			WithDiskType(string(apis.TypeSparseCPV)).
+			WithMaxPool(3).
+			WithOverProvisioning(false).
+			WithPoolType(string(apis.PoolTypeRaidz2CPV)).
+			Build()
+	})
+
+	// Test Case #1 (sparse-striped-auto-spc). | TestType : Pool Creation
+	When("We apply sparse-raidz2-auto spc yaml with maxPool count equal to 3 on a k8s cluster having at least 3 capable node", func() {
+		It("pool resources count should be 3 with no error and healthy status", func() {
+			// Create a storage pool claim
+			_, err := clientInstance.spcClient.Create(spcObj)
+			Expect(err).To(BeNil())
+			cspCount := isHealthyCspCount(spcObj.Name, 3)
+			Expect(cspCount).To(Equal(3))
+		})
+	})
+
+	//Test Case #2 : Dependent on above test case #1 . | TestType : Reconciliation
+	When("We delete 1 pool out of 3 by deleting one of the csp", func() {
+		It("a new pool should come up again", func() {
+			clientInstance.deleteCSP(spcObj.Name, 1)
+			// We expect 3 cstorPool objects.
+			cspCount := isHealthyCspCount(spcObj.Name, 3)
+			Expect(cspCount).To(Equal(3))
+		})
+	})
+
+	//Test Case #3 : Dependent on above test case #1 . | TestType : Reconciliation
+	When("We delete 2 pool out of 3 by deleting 2 csps", func() {
+		It("2 new pool should come up again", func() {
+			clientInstance.deleteCSP(spcObj.Name, 2)
+			// We expect 3 cstorPool objects.
+			cspCount := isHealthyCspCount(spcObj.Name, 3)
+			Expect(cspCount).To(Equal(3))
+		})
+	})
+
+	//Test Case #3 : Dependent on above test case #1 . | TestType : Reconciliation
+	When("We delete 3 pool out of 3 by deleting 3 csps", func() {
+		It("3 new pool should come up again", func() {
+			clientInstance.deleteCSP(spcObj.Name, 3)
+			// We expect 3 cstorPool objects.
+			cspCount := isHealthyCspCount(spcObj.Name, 3)
+			Expect(cspCount).To(Equal(3))
+		})
+	})
+
+})
+
+
+var _ = AfterSuite(func() {
+	spcClient, err := spc.KubeClient().WithFlag(KubeConfigPath)
+	spcList, err := spcClient.List(metav1.ListOptions{})
+	Expect(err).To(BeNil())
+	for _, spc := range spcList.Items {
+		_, err := spcClient.Delete(spc.Name, &metav1.DeleteOptions{})
+		Expect(err).To(BeNil())
+	}
+})
+
+var _ = BeforeSuite(func() {
+	for _, f := range clientBuilderFuncList {
+		f()
+	}
+})
+
+// clientInstacne is the initialized empty instance of client type.
+var clientInstance = &client{}
```

Comment:
I am viewing this `operation / ops`.
Can we change this to operation ?

Reply:
There can be cases where operation needs to include more than one clients e.g. `spcClient` & `cspClient` & so on. Hence my preference towards operation.

Reply:
make sense. Will fix it.